### PR TITLE
Add `MinutesTwoDigits` + `SecondsTwoDigits`

### PR DESCRIPTION
**Pull request by sectore**
To lead minutes and second with zeros if needed.

* Use `m` or `mm` pattern for formatting minutes
* Use `s` or `ss` pattern for formatting seconds
* Examples: 
    - `hh:m:s` -> `11:3:4`
    - `hh:mm:ss` -> `11:03:04`
* See https://momentjs.com/docs/#hour-minute-second-millisecond-and-offset-tokens

### CODE REVIEW

**Comment by sectore** [comment]
@garyb Any chance to review ⬆️? Thanks!

**Comment by garyb** [comment]
Sorry about the delay on this, I'll get to it soon!

File: src/Data/Formatter/DateTime.purs [hunk 1]
```diff
@@ -100,8 +104,10 @@ printFormatterF cb = case _ of
   Hours24 a → "HH" <> cb a
   Hours12 a → "hh" <> cb a
   Meridiem a → "a" <> cb a
-  Minutes a → "mm" <> cb a
-  Seconds a → "ss" <> cb a
+  Minutes a → "m" <> cb a
```
Comment (safareli) [inline src/Data/Formatter/DateTime.purs]: isn't it a breaking change?

Comment (garyb) [inline src/Data/Formatter/DateTime.purs]: It will be yep, we can roll it into the 0.11 updates.

File: src/Data/Formatter/DateTime.purs [hunk 2]
```diff
@@ -201,10 +209,14 @@ formatF cb dt@(DT.DateTime d t) = case _ of
...
   Minutes a →
     show (fromEnum $ T.minute t) <> cb a
+  MinutesTwoDigits a →
+    (padSingleDigit <<< fromEnum $ T.minute t) <> cb a
   Seconds a →
     show (fromEnum $ T.second t) <> cb a
+  SecondsTwoDigits a →
+    (padSingleDigit <<< fromEnum $ T.second t) <> cb a
   Milliseconds a →
-    show (fromEnum $ T.millisecond t) <> cb a
```
Comment (garyb) [inline src/Data/Formatter/DateTime.purs]: I don't know whether switching to `padSingleDigit` here is right or not - I don't think it's quite right though. I would have though maybe we need to pad it out to 3 digits (000-999) or not at all, so maybe this is another variety of option that needs adding too?

**Comment by garyb** [comment]
Looks great apart from the question about milliseconds formatting. Thanks!

File: src/Data/Formatter/DateTime.purs (hunk 1 above)
Comment (sectore) [inline src/Data/Formatter/DateTime.purs]: @garyb For me it is just a fix, but not a breaking change. Comparing to `moment.js` `ss` and `mm` do mean two digit values, not a single one. And `purescript-formatters` did not support this in a proper way before. With this pull request this issue would be fixed. 

It would be nice to to have this pull request before any `0.11` updates. Because it does not depend on any new feature of `0.11`. But using `0.11` does mean really a breaking change...

File: src/Data/Formatter/DateTime.purs (hunk 2 above)
Comment (sectore) [inline src/Data/Formatter/DateTime.purs]: @garyb Yep, it would be cool to pad milliseconds up to 3 digits, similar as `momentjs` it does by using formats of `S` `SS` `SSS` (https://momentjs.com/docs/#hour-minute-second-millisecond-and-offset-tokens) 

However, the goal of this pull request was just fixing the format of minutes / seconds, but not adding this extra feature. Please let do this with another pull request, I'm willing to do this if there a need. Thx!

File: src/Data/Formatter/DateTime.purs (hunk 1 above)
Comment (garyb) [inline src/Data/Formatter/DateTime.purs]: I hear what you're saying, but unfortunately it is a breaking change, as the ADT has been modified and its constructors are exported.

I can make a release for 0.10 before updating to 0.11 though, if you need this and are sticking with 0.10 for now. Just it will have its major version bumped too :)

File: src/Data/Formatter/DateTime.purs (hunk 2 above)
Comment (garyb) [inline src/Data/Formatter/DateTime.purs]: Sure, can we revert this change in this PR then?

Comment (sectore) [inline src/Data/Formatter/DateTime.purs]: Done.

File: src/Data/Formatter/DateTime.purs (hunk 1 above)
Comment (sectore) [inline src/Data/Formatter/DateTime.purs]: Cool, let's do this :)

**Comment by garyb** [comment]
Thanks! Sorry this took so long to get resolved.
